library/std/src/sys/unix/fs.rs

```diff
@@ -636,7 +636,7 @@ impl Iterator for ReadDir {
 impl Drop for Dir {
     fn drop(&mut self) {
         let r = unsafe { libc::closedir(self.0) };
-        debug_assert_eq!(r, 0);
+        assert_eq!(r, 0, "unexpected error during closedir: {:?}", r);
```

Comment:
I pointed out here that we should permit certain errors here: https://github.com/rust-lang/rust/issues/94705#issuecomment-1062086068.  Maybe all errors, but certainly at least `EINTR` should be ignored.

Reply:
>Maybe all errors, but certainly at least EINTR should be ignored.

Let's go in this way!

Reply:
with the following assert 
```
let r = unsafe { libc::closedir(self.0) }; 
assert_eq!(r != 0 && r != libc::EINTR,  "unexpected error during closedir: {:?}", r); 
```
A lot of tests will fail, do you think that this can be a good issue where start the discussion if this change really matters?

Reply:
Do you mean `r == 0 || r == EINTR`?

Reply:
>Do you mean r == 0 || r == EINTR?

Yeah! dummy error, sorry :) after rereading your review I realized